common/aws/test/client_test.go

```diff
@@ -0,0 +1,176 @@
+package test
+
+import (
+	"context"
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/aws"
+	"github.com/Layr-Labs/eigenda/common/aws/s3"
+	"github.com/Layr-Labs/eigenda/common/mock"
+	tu "github.com/Layr-Labs/eigenda/common/testutils"
+	"github.com/Layr-Labs/eigenda/inabox/deploy"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/assert"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+var (
+	dockertestPool     *dockertest.Pool
+	dockertestResource *dockertest.Resource
+)
+
+const (
+	localstackPort = "4570"
+	localstackHost = "http://0.0.0.0:4570"
+	bucket         = "eigen-test"
+)
+
+type clientBuilder struct {
+	// This method is called at the beginning of the test.
+	start func() error
+	// This method is called to build a new client.
+	build func() (s3.Client, error)
+	// This method is called at the end of the test when all operations are done.
+	finish func() error
+}
+
+var clientBuilders = []*clientBuilder{
```

Comment:
This is nice! The idea is to test every test against a mock (unit) and localstack (integration)?

Reply:
Yup! It's kind of like a meta-test to sanity check the unit tests. Since the mock implementation is super simple and easy to reason about, it helps me to distinguish between bugs in the unit test definition and bugs in the implementation of the code.